onnxruntime/contrib_ops/cpu/reverse.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/common/common.h"
+#include "core/framework/op_kernel.h"
+#include <unordered_set>
+#include "core/util/eigen_common_wrapper.h"
+
+namespace onnxruntime {
+namespace contrib {
+
+// some templated aliases
+template <typename T, int rank>
+using EigenTensorMap = Eigen::TensorMap<Eigen::Tensor<T, rank, Eigen::RowMajor, Eigen::DenseIndex>>;
+
+template <typename T, int rank>
+using ConstEigenTensorMap = Eigen::TensorMap<Eigen::Tensor<const T, rank, Eigen::RowMajor, Eigen::DenseIndex>>;
+
+// utility helpers specific to Reverse
+template <int rank>
+Eigen::array<bool, rank> vector_to_eigen_array(const std::vector<int64_t>& reverse_axes) {
```

Comment:
>vector_to_eigen_array [](start = 25, length = 21)

Name isn't really representative of what this does as it's not a generic conversion of a vector to an eigen array is it?

Reply:
How about `generate_bool_reverse_axes(...)` ?

Reply:
Yeah - that's much clearer